### PR TITLE
Add petrol bomb and fire

### PR DESCRIPTION
**Pull request by trundlr**
- Add petrol bomb weapon (needs sound pass)
- Add fire (needs sound pass)
- Crates explode when damaged
- All weapon components that do damage are updated to include crates
- Increase trace size for spawning so crates don't get stuck in wall

### CODE REVIEW

File: code/Terrain/FireHelper.cs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	}
+
+	public override void Spawn()
+	{
+		FireParticle = Particles.Create( "particles/fire/fire_base.vpcf", this, true );
+		Tags.Add( "fire" );
+		Name = "fire";
+	}
+
+	[GameEvent.Tick.Server]
```
Comment (matekdev) [inline code/Terrain/FireHelper.cs:50]: Will we have any problems moving the fire only on the server? Would it make sense to throw it into simulate?

File: code/Weapons/Gadget/Gadget.cs [hunk 2]
```diff
@@ -23,6 +23,8 @@ public partial class Gadget : AnimatedEntity, IResolvable
 
 	public bool Resolved => IsResolved();
 
+	public bool IsCrateGadget => Components.GetAll<GadgetComponent>().Any( c => c is CrateGadgetComponent );
```
Comment (matekdev) [inline code/Weapons/Gadget/Gadget.cs:26]: `public bool IsCrate => Components.Get<CrateGadgetComponent>() is not null;`
Does this work instead?

File: code/Terrain/FireHelper.cs (hunk 1 above)
Comment (trundlr) [reply code/Terrain/FireHelper.cs:50]: Works totally fine on the server - the movement is pretty incremental anyway for whatever reason and the move controller does not get affected by fire so it shouldn't affect prediction.

File: code/Weapons/Gadget/Gadget.cs (hunk 2 above)
Comment (trundlr) [reply code/Weapons/Gadget/Gadget.cs:26]: Good catch 👍

File: code/Terrain/ExplosionHelper.cs [hunk 3]
```diff
@@ -11,31 +11,34 @@ public static partial class ExplosionHelper
...
+			var dist = Vector3.DistanceBetween( position, entity.Position );
 			if ( dist > radius )
 				continue;
 
 			var distanceFactor = 1.0f - Math.Clamp( dist / radius, 0, 1 );
 			// TODO: PhysicsGroup/Body is invalid on grubs
 			var force = distanceFactor * 1000;
 
-			var dir = (grub.Position - position).Normal;
+			var dir = (entity.Position - position).Normal;
```
[the remark this answers is not in the document]
Comment (trundlr) [reply code/Terrain/ExplosionHelper.cs:32]: Good catch

**Pull request merged**